### PR TITLE
Ignore exhaustiveness check for some semantics tests, for now

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
~This also includes the engine hash change from https://github.com/flutter/flutter/pull/159453/files for testing purposes. Will revert if all tests pass~ all tests passed with the engine hash from #159453: https://github.com/flutter/flutter/pull/159517/commits/0f26acb438024948ddf5ea2cc7a17d1ed9f15193

TODOs: https://github.com/flutter/flutter/issues/159515
context: https://github.com/flutter/flutter/issues/159456

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by LongCatIsLooong** [comment]
also /cc @gaaclarke

File: packages/flutter/test/widgets/custom_painter_test.dart [hunk 1]
```diff
@@ -356,6 +356,7 @@ void _defineTests() {
     final Set<SemanticsAction> allActions = SemanticsAction.values.toSet()
       ..remove(SemanticsAction.customAction) // customAction is not user-exposed.
       ..remove(SemanticsAction.showOnScreen) // showOnScreen is not user-exposed
+      // TODO(LongCatIsLooong): https://github.com/flutter/flutter/issues/159515.
```
Comment (gaaclarke) [inline packages/flutter/test/widgets/custom_painter_test.dart:359]: What do these TODO's refer to?  I believe the format should be:

```c++
// TODO(159515): This is the thing we should do.
```

Comment (LongCatIsLooong) [reply packages/flutter/test/widgets/custom_painter_test.dart:359]: Per the style guide: https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-checking-in-comments-that-ask-questions

I'll add "clean up after `SemanticsAction.scrollToOffset` is added in dart:ui" to the description.

Comment (gaaclarke) [reply packages/flutter/test/widgets/custom_painter_test.dart]: As someone that is not you, I'm not sure what cleaning up entails.  There isn't a clear action that someone other than you could take.  A verb phase like "remove this" is more descriptive than "clean up".

**Review by gaaclarke — APPROVED** [review]
thanks =)

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**